desktop/src/com/frostwire/gui/bittorrent/CreateTorrentDialog.java

```diff
@@ -286,6 +321,21 @@ public void mouseClicked(MouseEvent e) {
                 }
             }
         });
+
+        pieceSizeComboBox.addActionListener(
+                new ActionListener() {
+                    @Override
+                    public void actionPerformed(ActionEvent e) {
+                        onPieceSizeSelected(pieceSizeComboBox.getSelectedIndex());
+                    }
+                }
+        );
+    }
+
+    private void onPieceSizeSelected(int selectedIndex) {
+        if (selectedIndex > 0 && values.length > selectedIndex && values[selectedIndex] != null) {
+            pieceSize = values[selectedIndex].kb*_1024BYTES;
```

Comment:
another way of doing this, would be to encapsulate the logic of how many bytes this is supposed to be, and not put the responsability on others to do this multiplication if you added a `.bytes()` method in the enum, which would then do the multiplication internally. I think that'd be cleaner.

Then here you'd have

```java
pieceSize = values[selectedIndex].bytes()
```